### PR TITLE
[qt5-base] Move 'icu' out of 'core' feature

### PR DESCRIPTION
**Pull request by Osyotr**
Extracted changes related to `icu` from #23668 cc. @dg0yt 

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/qt5-base/vcpkg.json [hunk 1]
```diff
@@ -44,6 +40,15 @@
     "zstd"
   ],
   "features": {
+    "icu": {
```
Comment (Osyotr) [inline ports/qt5-base/vcpkg.json:43]: I'm not sure whether `icu` should be default feature or not. It was hard dependency in vcpkg for many years, but it's not default in qt.

Comment (Neumann-A) [reply ports/qt5-base/vcpkg.json:43]: if it is not upstream default i would remove it. (unless there is some technical aspect which says: with icu is better and most of the times wanted)

**Review by dg0yt — APPROVED** [review]
LGTM. Thanks.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
Feature `icu` tested successfully by the following triplet:
- x86-windows
- x64-windows
- x64-windows-static

**Pull request merged**